### ISSUE TITLE
Fetching api version for store always shows store has wrong api version

### ISSUE THREAD

**Issue by AmandaRiu** [woocommerce/woocommerce-android#691]
**Describe the bug**
If an error occurs while fetching the WooCommerce store version from the API, the **Store Selector** screen will display that store as not having the correct API version (even though it may). Here is a snippet of the original code:

```
@Suppress("unused")
@Subscribe(threadMode = ThreadMode.MAIN)
fun onApiVersionFetched(event: OnApiVersionFetched) {
    if (event.apiVersion == WooCommerceStore.WOO_API_NAMESPACE_V3) {
        supportedWCSites.add(event.site)
    } else {
        unsupportedWCSites.add(event.site)
    }

    val totalSitesChecked = supportedWCSites.size + unsupportedWCSites.size
    if (totalSitesChecked == wooCommerceStore.getWooCommerceSites().size) {
        loginEpilogueView?.showStoreList(supportedWCSites, unsupportedWCSites)
    }
}
```

if `event.isError == true`, the `event.apiVersion` will be set to an empty string, so `event.apiVersion != WooCommerceStore.WOO_API_NAMESPACE_V3` and therefore the store will show up as having the wrong api version. The fix will be to add some proper error handling.

**Comment by melindahelt** [comment]
1725540-zen reported this issue. follow up with ticket when this is resolved

**Comment by aforcier** [comment]
Fixed in #695.
